core/src/types/token.rs

```diff
@@ -914,6 +1046,24 @@ pub fn is_masp_key(key: &Key) -> bool {
                     || key.starts_with(PIN_KEY_PREFIX)))
 }
 
+/// The last locked ratio of a token
+pub fn masp_last_locked_ratio(token_address: &Address) -> Key {
```

Comment:
Same as above regarding adding "_key" to the name. Also maybe change the docstring too